--- a/target/linux/ar71xx/image/Makefile
+++ b/target/linux/ar71xx/image/Makefile
@@ -1189,7 +1189,7 @@ $(eval $(call SingleProfile,TPLINK-LZMA,
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,ARCHERC7V2,archer-c7-v2,ARCHER-C7,ttyS0,115200,0xc7000002,1,16Mlzma))
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,ELM150,el-m150,EL-M150,ttyATH0,115200,0x01500101,1,8Mlzma))
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,ELMINI,el-mini,EL-MINI,ttyATH0,115200,0x01530001,1,8Mlzma))
-$(eval $(call SingleProfile,TPLINK-LZMA,64kraw,GLINET,gl-inet-v1,GL-INET,ttyATH0,115200,0x08000001,1,8Mlzma))
+$(eval $(call SingleProfile,TPLINK-LZMA,64kraw,GLINET,gl-inet-v1,GL-INET,ttyATH0,115200,0x08000001,1,16Mlzma))
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,TLMR10U,tl-mr10u-v1,TL-MR10U,ttyATH0,115200,0x00100101,1,4Mlzma))
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,TLMR11UV1,tl-mr11u-v1,TL-MR11U,ttyATH0,115200,0x00110101,1,4Mlzma))
 $(eval $(call SingleProfile,TPLINK-LZMA,64kraw,TLMR11UV2,tl-mr11u-v2,TL-MR11U,ttyATH0,115200,0x00110102,1,4Mlzma))
--- a/tools/firmware-utils/src/mktplinkfw.c
+++ b/tools/firmware-utils/src/mktplinkfw.c
@@ -198,13 +198,13 @@ static struct flash_layout layouts[] = {
 		.rootfs_ofs	= 0x140000,
 	}, {
 		.id		= "16Mlzma",
-		.fw_max_len	= 0xf80000,
+		.fw_max_len	= 0xfc0000,
 		.kernel_la	= 0x80060000,
 		.kernel_ep	= 0x80060000,
 		.rootfs_ofs	= 0x100000,
 	}, {
 		.id		= "16Mppc",
-		.fw_max_len	= 0xf80000,
+		.fw_max_len	= 0xfc0000,
 		.kernel_la	= 0x00000000,
 		.kernel_ep	= 0xc0000000,
 		.rootfs_ofs	= 0x2a0000,
@@ -398,7 +398,7 @@ static struct board_info boards[] = {
 		.id		= "GL-INETv1",
 		.hw_id		= HWID_GL_INET_V1,
 		.hw_rev		= 1,
-		.layout_id	= "8Mlzma",
+		.layout_id	= "16Mlzma",
 	}, {
 		.id		= "GS-OOLITEv1",
 		.hw_id		= HWID_GS_OOLITE_V1,
